capstone/src/main/java/com/google/sps/servlets/PollServlet.java

```diff
@@ -16,29 +15,49 @@
 import com.google.appengine.api.datastore.PreparedQuery;
 import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.datastore.Query.SortDirection;
-import com.google.appengine.api.users.UserService;
-import com.google.appengine.api.users.UserServiceFactory;
 import com.google.sps.Objects.Option;
 import com.google.sps.Objects.comparator.OptionsComparator;
+import com.google.sps.Objects.response.OptionsAndUserVotedOptions;
 import com.google.sps.Objects.response.PollResponse;
 
-import error.ErrorHandler;
-
 @WebServlet("/poll")
-public class PollServlet extends HttpServlet {
+public class PollServlet extends AuthenticatedServlet {
 
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String userId = "";
-    UserService userService = UserServiceFactory.getUserService();
-    if (userService.isUserLoggedIn()) {
-      userId = userService.getCurrentUser().getUserId();
-    } else {
-      ErrorHandler.sendError(response, "User is not logged in.");
-    }
+  public void doGet(String userId, HttpServletRequest request, HttpServletResponse response)
+      throws IOException {
     Query query = new Query("Option").addSort("timestamp", SortDirection.ASCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
+    OptionsAndUserVotedOptions optionsAndVotedOptions =
+        this.getOptionsAndVotedOptions(results, userId);
+    List<Option> options = optionsAndVotedOptions.getOptions();
+    List<Long> votedOptions = optionsAndVotedOptions.getVotedOptions();
+    // Sort list of options based on number of votes
+    Collections.sort(options, new OptionsComparator());
+    PollResponse pollResponse = new PollResponse(options, votedOptions, userId);
```

Comment:
I want to apologize since I know we talked about this during the our daily meeting, however I think we had a lack of context when talking about this. The OptionsAndVotedOptions here is kind of redundant since they are just used to pass into & create a PollResponse. So instead of needing to create the OptionsAndUserVotedOptions object, your helper method should create & return a PollResponse. 
Instead of: 
`
private OptionsAndUserVotedOptions getOptionsAndVotedOptions(PreparedQuery results, String userId) {
     ...
     return new OptionsAndUserVotedOptions(options, votedOptions);
}
`
you can have 
`
private PollResponse buildPollResponse(PreparedQuery results, String userId) {
     ...
     return new PollResponse(options, votedOptions, userId);
}
`
I know you also need to sort the options list so you can just add that in the helper method after building the list.
So that would leave you with:
`
PollResponse pollResponse = buildPollResponse(results, userId);
`